src/plasma-cash/ethereum-client.ts

```diff
@@ -174,6 +183,43 @@ export class EthereumPlasmaClient {
       .send(rest)
   }
 
+  /**
+   * `Invalid History Challenge`: Challenge a coin with invalid history.
+   *
+   * @returns Web3 tx receipt object.
+   */
+  challengeBeforeAsync(params: IPlasmaChallengeBeforeParams): Promise<object> {
+    const { slot, challengingTx, challengingBlockNum, prevTx, prevBlockNum, from, gas, gasPrice } = params
+    const prevTxBytes = prevTx ? prevTx.rlpEncode() : '0x'
+    const challengingTxBytes = challengingTx.rlpEncode()
+    const bond = this._web3.utils.toWei('0.1', 'ether')
+
+    return this._plasmaContract.methods
+      .challengeBefore(
+        slot,
+        prevTxBytes,
+        challengingTxBytes,
+        prevTx ? prevTx.proof : '0x',
+        challengingTx.proof,
+        challengingTx.sig,
+        [prevBlockNum || 0, challengingBlockNum]
+      )
+      .send({ from, value: bond, gas, gasPrice })
+  }
+
+  /**
+   * `Response to invalid history challenge`: Respond to an invalid challenge with a later tx
+   *
+   * @returns Web3 tx receipt object.
+   */
+  respondChallengeBefore(params: IPlasmaChallengeParams): Promise<object> {
```

Comment:
`respondChallengeBefore` should be `respondChallengeBeforeAsync`

Reply:
done